src/types/ReduxTypes.ts

```diff
@@ -42,6 +42,16 @@ export interface ExperimentConfig {
   createAccountText: 'signUp' | 'getStarted' | 'createAccount'
 }
 
+export const asExperimentConfig = (asT => asOptional(asT, asT({})))(
+  asObject<ExperimentConfig>({
+    createAccountType: asOptional(asValue('light', 'full'), 'full'),
+    createAccountText: asOptional(
+      asValue('signUp', 'getStarted', 'createAccount'),
+      'createAccount'
+    )
+  })
+)
```

Comment:
Can you give the `asT => asOptional(asT, asT({}))` part a name and put it on it's own line? Maybe `asOptionalObject` or something.

OR, just drop the weird cleaner behavior, and just use normal `asObject` with `asOptional` on the properties. Then, call `asExperimentConfig({})` instead of `asExperimentConfig(undefined)`, since `asWhatever(undefined)` is pretty strange-looking. For the main login component, do `asExperimentConfig(props.experimentConfig ?? {})` to provide the default in a more obvious way.